novawallet/Modules/Notifications/Management/NotificationsManagementInteractor.swift

```diff
@@ -67,24 +75,40 @@ extension NotificationsManagementInteractor: NotificationsManagementInteractorIn
 
     func save(
         settings: LocalPushSettings,
+        topics: LocalNotificationTopicSettings,
         notificationsEnabled: Bool,
         announcementsEnabled: Bool
     ) {
         if syncService == nil {
             syncService = alertServiceFactory.createSyncService()
         }
 
+        let group = DispatchGroup()
+        group.enter()
+        group.enter()
+
         syncService?.save(settings: settings) { [weak self] in
             self?.settingsManager.notificationsEnabled = notificationsEnabled
             self?.settingsManager.announcements = announcementsEnabled
-            DispatchQueue.main.async {
-                self?.presenter?.didReceiveSaveCompletion()
-            }
+            group.leave()
+        }
+
+        if topicService == nil {
+            topicService = alertServiceFactory.createTopicService()
+        }
+
+        topicService?.save(settings: topics) {
```

Comment:
if topic service is nil then we don't leave. I would probably move group.enter to the if statement.

Reply:
we create topicService before